core/GlobalState.cc

```diff
@@ -717,12 +715,6 @@ ArgInfo &GlobalState::enterMethodArgumentSymbol(Loc loc, SymbolRef owner, NameRe
 }
 
 string_view GlobalState::enterString(string_view nm) {
-    DEBUG_ONLY(if (isInitialized) {
-        if (nm != "<" && nm != "<<" && nm != "<=" && nm != "<=>" && nm != ">" && nm != ">>" && nm != ">=") {
```

Comment:
this test has found a lot of subtle bugs and I'd like to retain it if possible.

Reply:
Shouldn't this `if` have filtered it out and allowed it?

Reply:
@DarkDimius can you tell us more about what it prevents and why it's done while parsing strings?

I'm happy to provide a better/cleaner fix but I'm not sure to understand what is done here and why?

Reply:
We have a lot of internal names of form `<something>` that's chosen with `<` and `>` as you can't make this into a valid ruby identifier without suffering.
We want to make sure we don't round-trip through `string` for those names.